include/boost/math/special_functions/fibonacci.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020, Madhur Chauhan
+
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_FIBO_HPP
+#define BOOST_MATH_SPECIAL_FIBO_HPP
+
+#include "boost/math/constants/constants.hpp"
+#include <boost/assert.hpp>
+#include <boost/math/policies/error_handling.hpp>
+#include <cmath>
+#include <limits>
+
+#ifdef _MSC_VER
+#pragma once
+#endif
+
+namespace boost {
+namespace math {
+
+template <typename T, class Policy>
+T fibonacci(unsigned long long n, const Policy &pol) {
```

Comment:
Should there be a static assert that `T` is an integral type?

Reply:
I think a static assert for `is_arithmetic` should be better instead as there's no harm in using floating-point types.

I only require +,-,* operations on the type. 

C++20 style would be to use concepts and requires expression here, I guess.